contributor/reference/stable-release.rst

```diff
@@ -0,0 +1,76 @@
+Stable release process
+======================
+
+Key Steps at Each Stage
+-----------------------
+
+Rocks
++++++
+
+Currently, the rocks are published directly to the registry when a modification is made.
+The Rocks mono-repository is maintains a stable branch from which new rocks are built and published.
+
+The process is as follows:
+
+* **backport**: A backport is proposed to the stable branch.
+* **approval**: The change is reviewed and approved.
+* **build**: A GitHub action automatically builds and publishes the rock to the registry.
+
+.. note::
+
+    A rock is not automatically re-assigned to a charm version, the associated charms must be re-built and published to the charm store.
+
+Snaps
++++++
+
+Snaps are automatically built every 5 hours when there are changes and published to the related edge channel. Promotion to stable channels is done manually after validation.
+
+The process is as follows:
+
+* **backport**: A backport is proposed to the stable branch
+* **approval**: The change is reviewed and approved.
+* **build**: Launchpad builds the snap and publishes it to the edge channel. (There might be a delay of up to 5 hours)
+* **testing**: The snap is tested in the edge channel.
+* **promote**:
+    * If the snap passes testing, it is promoted to the beta channel.
+    * If the snap passes testing in the beta channel, it is promoted to the candidate channel.
+    * If the snap passes testing in the candidate channel, it is promoted to the stable channel.
+
+Charms
+++++++
+
+Charms are automatically built and published to the charm store on each and every commit.
+
+The process is as follows:
```

Comment:
Add more description, e.g. "Publishing charms to the charm store follows the following stages:
* **backport**: ...
* **approval**: ...
* **build**: ...
* **testing**: ...
* **promote**: ...